.github/workflows/code_checks.yml

```diff
@@ -220,17 +285,100 @@ jobs:
         with:
           check_name: Test Results
           comment_mode: off
-          files: Test Results/*.xml
+          files: Test Results Group*/*.xml
           github_token: ${{ secrets.GITHUB_TOKEN }}
 
+      - name: Check test-failure label
+        if: ${{ needs.tests.result == 'success' && github.event_name == 'pull_request' }}
+        id: check-test-failure-label
+        run: |
+          if gh pr view ${{ github.event.pull_request.number }} --json labels --jq '.labels[].name' | grep -q "^test-failure$"; then
+            echo "has_label=true" >> $GITHUB_OUTPUT
+          else
+            echo "has_label=false" >> $GITHUB_OUTPUT
+          fi
+        env:
+          GH_TOKEN: ${{ github.token }}
+
+      - name: Remove Test Failure label
+        if: ${{ needs.tests.result == 'success' && github.event_name == 'pull_request' && steps.check-test-failure-label.outputs.has_label == 'true' }}
```

Comment:
This prevents it from throwing an error annotation on the job if the label doesn't exist.